src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1642,11 +1642,10 @@ def process_disk_or_snapshot_create_namespace(cmd, namespace):
                     from azure.cli.core.util import parse_proxy_resource_id
                     result = parse_proxy_resource_id(namespace.source_disk or namespace.source_snapshot)
                     source_info, _ = _get_disk_or_snapshot_info(cmd.cli_ctx, result['resource_group'], result['name'])
-                source_location = source_info.location
-                target_location = namespace.location if namespace.location \
-                    else get_default_location_from_resource_group(cmd, namespace)
```

Comment:
`get_default_location_from_resource_group` won't return location, it will set `namespace.location=resource_group.location` inside.

So if user doesn't specify `--location`, `traget_location` will get `None` from `get_default_location_from_resource_group`.

The correct logic for getting target location is to use `namespace.location` after `get_default_location_from_resource_group`